### PR TITLE
shape: rewrite extension

### PR DESCRIPTION
**Pull request by benoit-pierre**
Replace with a version generated form xcb-proto.

### CODE REVIEW

File: examples/shapewin.py [hunk 1]
```diff
@@ -165,7 +165,7 @@ def loop(self):
                                            self.sub_pm)
 
             # Shape has changed
-            elif e.type == self.d.extension_event.ShapeNotify:
+            elif e.type == shape.Event.Notify:
```
Comment (petli) [inline examples/shapewin.py]: How does this pick up the display-specific event ID for the extension event?  Looks to me like shape.Event.Notify will just be 0.

Comment (benoit-pierre) [inline examples/shapewin.py]: Good catch, looking into it.

Comment (petli) [inline examples/shapewin.py]: Perhaps adding the new autogenerated events scoped under the extension name?  E.g. `self.d.extension_event.Shape.NotifyEventData`

(Though that whole dynamic creation of classes and methods seems like a mistake 16 years later...  It should be rethought to allow extensions to be implemented as separate packages, which would also avoid the whole license issue - if that's still an issue when the extensions can be autogenerated.)

Comment (benoit-pierre) [inline examples/shapewin.py]: See amended code in #43.

**Pull request merged**